pkg/daemon/daemon.go

```diff
@@ -855,6 +856,20 @@ func (dn *Daemon) RunFirstbootCompleteMachineconfig() error {
 		return nil
 	}
 
+	client := rpmostreeclient.NewClient("machine-config-daemon")
+	newEnough, err := RpmOstreeIsNewEnoughForLayering(client)
+	if err != nil {
+		return err
+	}
+	// If the host isn't new enough to understand the new container model natively, run as a privileged container and reboot.
+	// See https://github.com/coreos/rpm-ostree/pull/3961 and https://issues.redhat.com/browse/MCO-356
+	if !newEnough {
+		err := runCmdSync("podman", "run", "--privileged", "--pid=host", "--net=host", "--rm", "-v", "/:/run/host", mc.Spec.OSImageURL, "rpm-ostree", "ex", "deploy-from-self", "/run/host", "--reboot")
```

Comment:
Just curious, this is basically updating the on-host rpm-ostree from the updated rpm-ostree in the updated (4.12) oscontainer. Why is the reboot needed vs, say, restarting the rpm-ostree daemon?

Reply:
The apply-live path [was only stabilized in 2022.13](https://github.com/coreos/rpm-ostree/pull/3907/commits/30bf326147774b2ebbaa5abf4f9e0d2cfdee2432)...earlier versions do have `ex apply-live`, but really old versions don't.

We could indeed optimize this in the future to skip a reboot for sufficiently new versions; but I think ultimately what we want to do is to enhance this code to do all the setup from the container too, to avoid the *next* update doing a reboot.

IOW our firstboot path then always runs rpm-ostree from a container, and not the host.

On the other hand, we have some code for updating bootimages, and once that lands this code is going to be hit much less often.